core/src/types/operator/operator_futures.rs

```diff
@@ -384,6 +395,16 @@ impl<F: Future<Output = Result<Writer>>> FutureWriter<F> {
     pub fn executor(self, executor: Executor) -> Self {
         self.map(|(args, options)| (args.with_executor(executor), options))
     }
+
+    /// Set the user defined metadata of the op
+    ///
+    /// ## Notes
+    ///
+    /// we don't need to include the user defined metadata prefix in the key
+    /// every service will handle it internally
+    pub fn user_metadata(self, data: HashMap<String, String>) -> Self {
```

Comment:
Let's accept an `FromIterator` to make it easier to use.

Reply:
Do you mean `IntoIterator` ?
like this:
```
    pub fn user_metadata(self, data: impl IntoIterator<Item = (String, String)>) -> Self {
        self.map(|(args, options, bs)| (args.with_user_metadata(HashMap::from_iter(data)), options, bs))
    }
```